test/test_main.py

```diff
@@ -620,6 +624,7 @@ def test_collect_metrics(self):
             VERSION_LOGLINE,
             ['juju-log', '--log-level', 'DEBUG', '--', 'Emitting Juju event collect_metrics.'],
             ['add-metric', '--labels', 'bar=4.2', 'foo=42'],
+            ['is-leader', '--format=json'],
```

Comment:
The testing of `_evaluate_status` is done in `test_charm.py`; all this has to do is test whether it's called.